pkg/cloudprovider/provider/kubevirt/types/types.go

```diff
@@ -19,20 +19,57 @@ package types
 import (
 	"github.com/kubermatic/machine-controller/pkg/jsonutil"
 	providerconfigtypes "github.com/kubermatic/machine-controller/pkg/providerconfig/types"
-
 	corev1 "k8s.io/api/core/v1"
 )
 
 type RawConfig struct {
-	Kubeconfig       providerconfigtypes.ConfigVarString `json:"kubeconfig,omitempty"`
-	CPUs             providerconfigtypes.ConfigVarString `json:"cpus,omitempty"`
-	Memory           providerconfigtypes.ConfigVarString `json:"memory,omitempty"`
-	Namespace        providerconfigtypes.ConfigVarString `json:"namespace,omitempty"`
-	SourceURL        providerconfigtypes.ConfigVarString `json:"sourceURL,omitempty"`
-	PVCSize          providerconfigtypes.ConfigVarString `json:"pvcSize,omitempty"`
+	Auth           Auth           `json:"auth,omitempty"`
+	VirtualMachine VirtualMachine `json:"virtualMachine,omitempty"`
+}
+
+// Auth
+type Auth struct {
+	Kubeconfig providerconfigtypes.ConfigVarString `json:"kubeconfig,omitempty"`
+}
+
+// VirtualMachine
+type VirtualMachine struct {
+	Name      providerconfigtypes.ConfigVarString `json:"name,omitempty"`
+	Flavor    Flavor                              `json:"flavor,omitempty"`
+	Template  Template                            `json:"template,omitempty"`
+	DNSPolicy providerconfigtypes.ConfigVarString `json:"dnsPolicy,omitempty"`
+	DNSConfig *corev1.PodDNSConfig                `json:"dnsConfig,omitempty"`
+}
+
+// Flavor
+type Flavor struct {
+	Name    providerconfigtypes.ConfigVarString `json:"name,omitempty"`
+	Profile providerconfigtypes.ConfigVarString `json:"profile,omitempty"`
+}
+
+// Template
+type Template struct {
+	CPUs           providerconfigtypes.ConfigVarString `json:"cpus,omitempty"`
+	Memory         providerconfigtypes.ConfigVarString `json:"memory,omitempty"`
+	PrimaryDisk    PrimaryDisk                         `json:"primaryDisk,omitempty"`
```

Comment:
Why not just having Disks, instead of Primary disk and secondary disks?

Reply:
Here is the rawConfig definition from Marcin: https://kubermatic.slack.com/archives/C01JHBHE82F/p1638872204002600

`Resources` inside `DataVolumeTemplateSpec` for root disk comes from PrimaryDisk, so having primary and secondary disks as separate helps in identifying the resources.